### PR TITLE
feat(showcase): added showcase and refactored

### LINKED ISSUE #395: stark-build: get rid of the old webpack way to execute tests with Karma
Goal: remove https://github.com/NationalBankBelgium/stark/blob/master/packages/stark-build/config/karma.conf.js and its usages as these are not used anymore.

Also clean up the old stark-old configurations

### PR DESCRIPTION
**Pull request by dsebastien**
Closes #395 #353 
Contributes to #130 and #63

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
* no showcase app
* polluted starter app
* legacy karma testing config

## What is the new behavior?
* showcase added
* started the starter cleanup
* removed old karma testing config

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: showcase/assets/translations/en.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "SHOWCASE": {
+    "HOME": {
+      "TITLE": "Stark Starter"
```
Comment (RobbyDeLaet) [inline showcase/assets/translations/en.json]: Shouldn't we call this one "Showcase" instead of "Starter"

Comment (dsebastien) [reply showcase/assets/translations/en.json]: I'll adapt the PR and also take the occasion to rebase

File: showcase/angular.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    "showcase-e2e": {
+      "root": "",
+      "sourceRoot": "",
+      "projectType": "application"
+    }
+  },
+  "defaultProject": "showcase",
+  "schematics": {
+    "@schematics/angular:component": {
+      "prefix": "app",
```
Comment (christophercr) [inline showcase/angular.json]: Shouldn't this prefix be 'showcase'?

File: showcase/angular.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      "styleext": "pcss"
```
Comment (christophercr) [inline showcase/angular.json]: I think we will use scss files now to align with angular material...

File: showcase/angular.json [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    },
+    "@schematics/angular:directive": {
+      "prefix": "app"
```
Comment (christophercr) [inline showcase/angular.json]: Shouldn't this prefix be 'showcase'?

File: showcase/assets-base/manifest.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "NBB Stark",
```
Comment (christophercr) [inline showcase/assets-base/manifest.json]: Showcase?

File: package.json [hunk 6]
```diff
@@ -66,74 +66,90 @@
...
+    "docs:coverage": "npm run docs:stark-core:coverage && npm run docs:stark-ui:coverage && npm run docs:starter:coverage && npm run docs:showcase:coverage",
     "docs:stark-core:coverage": "cd packages/stark-core && npm run docs:coverage && cd ../..",
     "docs:stark-core:generate": "cd packages/stark-core && npm run docs && cd ../..",
     "docs:stark-core:serve": "cd packages/stark-core && npm run docs:serve && cd ../..",
     "docs:stark-ui:coverage": "cd packages/stark-ui && npm run docs:coverage && cd ../..",
     "docs:stark-ui:generate": "cd packages/stark-ui && npm run docs && cd ../..",
     "docs:stark-ui:serve": "cd packages/stark-ui && npm run docs:serve && cd ../..",
+    "docs:showcase:coverage": "cd showcase && npm run docs:coverage && cd ..",
     "docs:starter:coverage": "cd starter && npm run docs:coverage && cd ..",
+    "docs:showcase:generate": "cd showcase && npm run docs && cd ..",
```
[the remark this answers is not in the document]
Comment (dsebastien) [reply package.json]: Nope, good point, I'll drop that

File: showcase/angular.json (hunk 2 above)
Comment (dsebastien) [reply showcase/angular.json]: Sure why not, I'll adapt

File: showcase/angular.json (hunk 4 above)
Comment (dsebastien) [reply showcase/angular.json]: Yes, changing

File: showcase/assets-base/manifest.json (hunk 5 above)
Comment (dsebastien) [reply showcase/assets-base/manifest.json]: Ok, changing

File: showcase/angular.json (hunk 3 above)
Comment (christophercr) [reply showcase/angular.json]: sure, indeed it is a major change

File: showcase/src/index.html [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+	<meta charset="utf-8">
+	<meta name="description" content="<%= htmlWebpackPlugin.options.starkAppMetadata.description %>">
+	<meta name="viewport" content="width=device-width, initial-scale=1">
+	<!-- Disable tap highlight on IE -->
+	<meta name="msapplication-tap-highlight" content="no">
+	<!-- Add to homescreen -->
+	<link rel="manifest" href="manifest.json">
+	<!-- Fallback to homescreen for Chrome <39 on Android -->
+	<meta name="mobile-web-app-capable" content="yes">
+	<meta name="application-name" content="Stark Demo">
```
[the remark this answers is not in the document]
Comment (dsebastien) [reply showcase/src/index.html]: Yup, changing

**Comment by dsebastien** [comment]
Ok, changes pushed to my fork

**Pull request merged**